---
title: "Genome annotation"
subtitle: "Microbiomes2021"
author: "Sandra Godinho Silva"
institute: "Instituto Superior Técnico"
date: "(updated: `r Sys.Date()`)"
output:
  xaringan::moon_reader:
    seal: false
    lib_dir: libs
    css: ["xaringan-themer.css", "slides-style.css"]
    nature:
      highlightStyle: solarized-light
      highlightLines: true
      countIncrementalSlides: false
      ratio: "16:9"
      slideNumberFormat: |
        <div class="progress-bar-container">
          <div class="progress-bar" style="width: calc(%current% / %total% * 100%);">
          </div>
        </div>
---
```{r meta, echo=FALSE}
library(metathis)
meta() %>%
  meta_general(
    description = "Intro",
    generator = "xaringan and remark.js"
  ) %>% 
  meta_name("github-repo" = "sandragodinhosilva/microbiomes2021") %>% 
  meta_social(
    title = "Intro 1",
    url = "https://sandragodinhosilva/microbiomes2021/slides/day-01-basics.html#1",
    image = "https://raw.githubusercontent.com/sandragodinhosilva/microbiomes2021/main/slides/day-01-cover-image.png",
    image_alt = "Title slide",
    og_type = "website",
    og_author = "Sandra Godinho Silva",
    twitter_card_type = "summary_large_image",
    twitter_creator = "@SandraGodSilva"
  )
```
```{r set-theme, include=FALSE}
library(xaringanthemer)
style_duo_accent(
  primary_color      = "#0F4C81", # pantone classic blue
  secondary_color    = "#B6CADA", # pantone baby blue
  header_font_google = google_font("Raleway"),
  text_font_google   = google_font("Raleway", "300", "300i"),
  code_font_google   = google_font("Source Code Pro"),
  text_font_size     = "30px"
)
```

```{r setup, include = FALSE}
options(htmltools.dir.version = FALSE)
library(knitr)
library(broom)
library(scales)
library(openintro)
library(jsonlite)
library(tidyverse)
library(xaringanExtra)
library(nhsrtheme)
library(NHSRdatasets)
# set default options
opts_chunk$set(echo=FALSE,
               collapse = TRUE,
               fig.width = 7.252,
               fig.height = 4,
               dpi = 300)
# set engines
knitr::knit_engines$set("markdown")
xaringanExtra::use_tile_view()
xaringanExtra::use_panelset()
xaringanExtra::use_clipboard()
xaringanExtra::use_webcam()
xaringanExtra::use_broadcast()
xaringanExtra::use_share_again()
xaringanExtra::style_share_again(
  share_buttons = c("twitter", "linkedin", "pocket")
)

```
class: title-slide, right, top
background-image: url(img/dna.html)


.right-column[
# `r rmarkdown::metadata$title`
### `r rmarkdown::metadata$subtitle`

**`r rmarkdown::metadata$author`**<br>
`r rmarkdown::metadata$date`
]

---
name: about-me
class: center, middle


## About me: Sandra Godinho Silva

<img style="border-radius: 50%;" src="https://sandragodinhosilva.netlify.app/authors/admin/avatar_hu27f7dc34c9db8fd2b063f3fba6be9864_589773_270x270_fill_q75_lanczos_center.jpg"/>

## 

### PhD Student @ Institute for Bioengineering and Biosciences

.fade[Instituto Superior Técnico<br>Lisbon, Portugal]

---



class: inverse, center, middle

# Get Started: Bioinformatics



---

# Data Science
```{r datascience, echo=FALSE, out.width="75%", fig.align="center"}
knitr::include_graphics("img/data-analysis.jpg")
```

.footnote[Caption: Allison Horst [@allison_horst](https://twitter.com/allison_horst)
Adapted from Wickham & Grolemund 2017
]
---


# Bash
* Default login shell for most Linux distributions;
* Simple bash scripts are really useful for data manipulation.


.pull-left[
```{r bash, echo=FALSE, out.width="60%"}
knitr::include_graphics("img/bash.png")
```
]

---

# Python
* Clear and powerful object-oriented programming language;
* Portable – runs just about anywhere;
* Clear syntax – relatively easy to learn.

.pull-left[
```{r python, echo=FALSE, out.width="60%"}
knitr::include_graphics("img/python.png")
```
]

.pull-right[
.midi[Source: [https://www.python.org/](https://www.python.org)]
]


---

# Biopython

* Collection of .blue[Python] modules that provide functions to deal with DNA, RNA & protein sequence operations;
* It has sibling projects like BioPerl, BioJava and BioRuby.

.pull-left[
```{r biopython, echo=FALSE, out.width="70%"}
knitr::include_graphics("img/Biopython_logo.svg")
```
]

.pull-right[
.midi[Source: [https://biopython.org/](https://biopython.org)
]
]


---

#R

* .purple[R] is a powerful, popular open-source scripting language
* More than 20 years old - fairly mature - and growing in popularity. 

.pull-left[
```{r rlogo, echo=FALSE, out.width="70%"}
knitr::include_graphics("img/Rlogo.png")
```
]
.pull-right[
.midi[Source: [https://www.r-project.org//](https://biopython.org)
]
]

---

# R Studio

* Integrated Development Environment (IDE). 
* Provides a graphical interface to R, making it more user-friendly, and providing dozens of useful features.

.pull-left[
```{r rstudio, echo=FALSE, out.width="100%"}
knitr::include_graphics("img/rstudio.png")
```
]
.pull-right[
.midi[Source: [https://biopython.org/](https://biopython.org)
]
]


---

# Why is R so popular?


---

# And much more!

```{r programming, echo=FALSE, out.width="70%",fig.align = "center"}
knitr::include_graphics("img/programming_languages.jpeg")
```



---

# A note on Reproducible research


How to ensure reproducibility:
* Automation (write script to do the tasks, avoid manual editing)

* Git version control code and data.

* Jupyter/R Notebook, documentation

* Containers (docker, singularity, biocontainers)
---

# Genome Annotation



---
## Introduction into File Formats
.panelset[

.panel[.panel-name[FASTA]

DNA and protein sequences are written in FASTA format where you have in the first line a ">" followed by the description. In the second line the sequence starts.

```{r fasta, echo=FALSE, out.width="50%",fig.align = "center"}
knitr::include_graphics("img/fasta.png")
```

]

.panel[.panel-name[GFF3]

The general feature format (gene-finding format, generic feature format, GFF) is a file format used for describing genes and other features of DNA, RNA and protein sequences.

```{r gff3, echo=FALSE, out.width="70%",fig.align = "center"}
knitr::include_graphics("img/gff3_format.png")
```
]


.panel[.panel-name[GENBANK]

The genbank sequence format is a rich format for storing sequences and associated annotations.

```{r genbank, echo=FALSE, out.width="50%",fig.align = "center"}
knitr::include_graphics("img/gb_full.png")
```

]
]


???
Source: https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html

--
---


# Functional Annotation

Functional gene annotation means the description of the biochemical and biological function of proteins. 

Possible analyses to annotate genes can be:

* similarity searches
* gene cluster prediction for secondary metabolites
* identification of transmembrane domains in protein sequences
* finding gene ontology terms
* pathway information


E.g.: For similarity searches we use NCBI BLAST+ blastp to find similar proteins in a protein database.

---

class: center, middle

# Thanks!

Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).
