---
title: "Bioinformatic tools for Genome annotation"
subtitle: "Course: Microbiomes (2021)"
author: "Sandra Godinho Silva"
institute: "Instituto Superior Técnico"
date: "-04-2021"
output:
  xaringan::moon_reader:
    seal: true
    lib_dir: libs
    css: ["xaringan-themer.css", "slides-style.css"]
    nature:
      highlightStyle: solarized-light
      highlightLines: true
      countIncrementalSlides: false
      ratio: "16:9"
      slideNumberFormat: |
        <div class="progress-bar-container">
          <div class="progress-bar" style="width: calc(%current% / %total% * 100%);">
          </div>
        </div>
---
```{r meta, echo=FALSE}
library(metathis)
meta() %>%
  meta_general(
    description = "Intro",
    generator = "xaringan and remark.js"
  ) %>% 
  meta_name("github-repo" = "sandragodinhosilva/microbiomes2021") %>% 
  meta_social(
    title = "Intro 1",
    url = "https://sandragodinhosilva/microbiomes2021/slides/day-01-basics.html#1",
    image = "https://raw.githubusercontent.com/sandragodinhosilva/microbiomes2021/main/slides/day-01-cover-image.png",
    image_alt = "Title slide",
    og_type = "website",
    og_author = "Sandra Godinho Silva",
    twitter_card_type = "summary_large_image",
    twitter_creator = "@SandraGodSilva"
  )
```
```{r set-theme, include=FALSE}
library(xaringanthemer)
style_duo_accent(
  primary_color      = "#0F4C81", # pantone classic blue
  secondary_color    = "#B6CADA", # pantone baby blue
  header_font_google = google_font("Raleway"),
  text_font_google   = google_font("Raleway", "300", "300i"),
  code_font_google   = google_font("Source Code Pro"),
  text_font_size     = "25px"
)
```

```{r setup, include = FALSE}
options(htmltools.dir.version = FALSE)
library(knitr)
library(broom)
library(scales)
library(openintro)
library(jsonlite)
library(tidyverse)
library(xaringanExtra)
library(nhsrtheme)
library(NHSRdatasets)
# set default options
opts_chunk$set(echo=FALSE,
               collapse = TRUE,
               fig.width = 7.252,
               fig.height = 4,
               dpi = 300)
# set engines
knitr::knit_engines$set("markdown")
xaringanExtra::use_tile_view()
xaringanExtra::use_panelset()
xaringanExtra::use_clipboard()
xaringanExtra::use_webcam()
xaringanExtra::use_broadcast()
xaringanExtra::use_share_again()
xaringanExtra::style_share_again(
  share_buttons = c("twitter", "linkedin", "pocket")
)

```

name: about-me
class: center, middle

## About me: Sandra Godinho Silva

<img style="border-radius: 50%;" src="https://sandragodinhosilva.netlify.app/authors/admin/avatar_hu27f7dc34c9db8fd2b063f3fba6be9864_589773_270x270_fill_q75_lanczos_center.jpg"/>

## 

### PhD Student @ Institute for Bioengineering and Biosciences

.fade[Instituto Superior Técnico<br>Lisbon, Portugal]

---
class:  center, middle

# Bioinformatics

Bioinformatics is the science that conjugates  
**biology** + **statistics** + **computer sciences**  
to study biological issues through the analysis of data.

---
class: middle

**Bioinformatics is a constantly changing and updating field.**

Fast developments in:
- software
- computing hardware 
- high throughput technologies 

--

are creating a significant bottleneck:  
**How to thoroughly analyze this massive amount of data?**

---
**How to thoroughly analyze this massive amount of data?**

```{r bigdata, echo=FALSE, out.width="70%", fig.align="center"}
knitr::include_graphics("img/bigdata_lifesciences.png")
```
.center[
[Stephens et a. PLoS Biology, 2015.](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002195)
]

---
## Integrative OMICS
```{r integrative, echo=FALSE, out.width="80%", fig.align="center"}
knitr::include_graphics("img/integrative_omics.png")
```
.center[
[ Yugi et al., Trends Biotechnol. 2016 Apr; 34(4):276–290](https://doi.org/10.1016/j.tibtech.2015.12.013)
]

---
## Choose the most appropriate route for you

Nowadays, bioinformaticians and researchers working in life scientists can choose from an overwhelming collection of exciting technologies and programming languages

Several options:
- Web-based platforms
- Scripting/programming

---
# Web-based platforms

Allow **accessible**, **reproducible**, and **transparent computational research.**

.panelset[

.panel[.panel-name[KBase]

.pull-left[
```{r kbase, echo=FALSE, out.width="60%"}
knitr::include_graphics("img/KBase-logo.png")
```
]
.pull-right[The U.S. Department of Energy Systems Biology Knowledgebase [KBase](https://www.kbase.us/)  is an open-source software and data platform designed to meet the grand challenge of systems biology — predicting and designing biological function from the biomolecular (small scale) to the ecological (large scale)
]]


.panel[.panel-name[Galaxy]
.pull-left[
```{r galaxy, echo=FALSE, out.width="60%"}
knitr::include_graphics("img/galaxy.png")
```
]
.pull-left[Main goal: to develop and maintain a system that enables researchers without informatics expertise to perform computational analyses through the web. A user interacts with Galaxy through the web by uploading and analyzing the data. Galaxy interacts with underlying computational infrastructure (servers that run the analyses and disks that store the data) without exposing it to the user.
(https://galaxyproject.org/)
]]
]


---
# Scripting & programming

.panelset[

.panel[.panel-name[Bash]
.pull-left[
```{r bash, echo=FALSE, out.width="60%"}
knitr::include_graphics("img/bash.png")
```
]
.pull-right[
* Default login shell for most Linux distributions;
* Simple bash scripts are really useful for data manipulation.
]
]

.panel[.panel-name[Python]
.pull-left[
```{r python, echo=FALSE, out.width="50%"}
knitr::include_graphics("img/python.png")
```
]
.pull-right[
* Clear and powerful object-oriented programming language;
* Portable – runs just about anywhere;
* Clear syntax – relatively easy to learn.
Source: [https://www.python.org/](https://www.python.org)]
]

.panel[.panel-name[Biopython]

.pull-left[
```{r biopython, echo=FALSE, out.width="70%"}
knitr::include_graphics("img/Biopython_logo.svg")
```
]
.pull-right[
* Collection of .blue[Python] modules that provide functions to deal with DNA, RNA & protein sequence operations;
* It has sibling projects like BioPerl, BioJava and BioRuby.
Source: [https://biopython.org/](https://biopython.org)
]
]

.panel[.panel-name[R]
.pull-left[
```{r rlogo, echo=FALSE, out.width="70%"}
knitr::include_graphics("img/Rlogo.png")
```
]
.pull-right[
* .purple[R] is a powerful, popular open-source scripting language
* More than 20 years old - fairly mature - and growing in popularity. 
]
]

]

---

# And much more!

```{r programming, echo=FALSE, out.width="70%",fig.align = "center"}
knitr::include_graphics("img/programming_languages.jpeg")
```


---
# Why is R so popular?



---

# R Studio

* Integrated Development Environment (IDE). 
* Provides a graphical interface to R, making it more user-friendly, and providing dozens of useful features.

.pull-left[
```{r rstudio, echo=FALSE, out.width="100%"}
knitr::include_graphics("img/rstudio.png")
```
]
.pull-right[
.midi[Source: [https://biopython.org/](https://biopython.org)
]
]


---
# Genome Annotation

The process of decorating the genome with information about the localization and function of genes  
The genome annotation procedure will also embrace other functional elements such as promoter sequences,
non-coding RNAs, among others.

---

```{r genome_annot, echo=FALSE, out.width="70%", fig.align="center"}
knitr::include_graphics("img/genome-annotation.jpg")
```
.midi[Source: https://www.ssbs.edu.in/genome-annotation.html]

???
Genome Annotation is the process to identify, analyze and interpret the relevant information related to the raw DNA sequences. Genome Annotation helps to extract structural location and biological functions of genes. Haemophilus influenza was the first free-living bacterium to be decoded in 1995 by Dr. Owen White. The assembling of the quality reads with a reference genome or de novo assembly in order to obtain the complete genome is a primary requirement for Genome Annotation.

---
## Introduction into File Formats
.panelset[

.panel[.panel-name[FASTA]

DNA and protein sequences can be written in FASTA format.
First line:">" followed by the description. 
In the second line the sequence starts.

```{r fasta, echo=FALSE, out.width="50%",fig.align = "center"}
knitr::include_graphics("img/fasta.png")
```

]

.panel[.panel-name[GFF3]

General feature format (gene-finding format, generic feature format, GFF) is a file format used for describing genes and other features of DNA, RNA and protein sequences.

```{r gff3, echo=FALSE, out.width="70%",fig.align = "center"}
knitr::include_graphics("img/gff3_format.png")
```
]


.panel[.panel-name[GENBANK]

The genbank sequence format is a rich format for storing sequences and associated annotations.

```{r genbank, echo=FALSE, out.width="50%",fig.align = "center"}
knitr::include_graphics("img/gb_full.png")
```

]
]


???
Source: https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html

--
---


# Functional Annotation

Functional gene annotation means the description of the biochemical and biological function of proteins. 

Possible analyses to annotate genes can be:

* similarity searches
* gene cluster prediction for secondary metabolites
* identification of transmembrane domains in protein sequences
* finding gene ontology terms
* pathway information


E.g.: For similarity searches we use NCBI BLAST+ blastp to find similar proteins in a protein database.

---

# Databases

.panelset[

.panel[.panel-name[PFAM]
.left-column[

.center[
```{r pfam, echo=FALSE, out.width="100%", fig.align="center"}
knitr::include_graphics("img/pfam.gif")
```
]
]

.right-column[
Pfam is a large collection of protein families, represented by multiple sequence alignments and hidden Markov models (HMMs).
]
]

.panel[.panel-name[COG]

.left-column[
# COG
]
.right-column[
**Clusters of Orthologous Groups (COGs)**

Database of proteins generated by comparing the protein sequences of complete genomes. Each cluster contains groups of paralogs from at least three lineages.

]
]

.panel[.panel-name[Kegg]
.left-column[
```{r kegg, echo=FALSE, out.width="100%"}
knitr::include_graphics("img/KEGG_database_logo.gif")
```
]
.right-column[
**Kyoto Encyclopedia of Genes and Genomes**

Database resource for understanding high-level functions and utilities of biological systems, such as the cell, the organism and the ecosystem, from molecular-level information.
]
]

.panel[.panel-name[CAZymes]
.left-column[
```{r cazymes, echo=FALSE, out.width="150%"}
knitr::include_graphics("img/cazymes.png")
```
]
.right-column[
**Carbohydrate-Active enZymes**

Enzymes involved in the synthesis, metabolism, and transport of carbohydrates. 

CAZymes are organized in families in the continuously updated database [CAZy](www.cazy.org).

Includes: glycoside hydrolases (GHs), glycosyltransferases (GTs), polysaccharide lyases (PLs), carbohydrate esterases (CEs) and carbohydrate binding modules (CBMs). 

]
]

]

---
# Tools for genome annotation


Prokka

[Tutorial](https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html)
---

```{r melange, echo=FALSE, out.width="100%"}
knitr::include_graphics("img/melange.jpeg")
```


Documentation: https://sandragodinhosilva.github.io/MeLanGE/
---
# Your turn!

Genome annotation with Protein Families Database (PFAM)

Tools: WebMGA + R 

http://weizhong-lab.ucsd.edu/webMGA/server/pfam/

---
# WebMGA
```{r webmga, echo=FALSE, out.width="55%", fig.align="center"}
knitr::include_graphics("img/webmga_pfam.png")
```
[Link](http://weizhong-lab.ucsd.edu/webMGA/server/)

---

## R script to join annotations from different MAGs



---

class: center, middle

# Thanks!

Here's where you can find me:


Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).
