<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Bioinformatic tools for Genome annotation</title>
    <meta charset="utf-8" />
    <meta name="author" content="Sandra Godinho Silva" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <meta name="description" content="Intro"/>
    <meta name="generator" content="xaringan and remark.js"/>
    <meta name="github-repo" content="sandragodinhosilva/microbiomes2021"/>
    <meta name="twitter:title" content="Intro 1"/>
    <meta name="twitter:description" content="Intro"/>
    <meta name="twitter:url" content="https://sandragodinhosilva/microbiomes2021/slides/day-01-basics.html#1"/>
    <meta name="twitter:image:src" content="https://raw.githubusercontent.com/sandragodinhosilva/microbiomes2021/main/slides/day-01-cover-image.png"/>
    <meta name="twitter:image:alt" content="Title slide"/>
    <meta name="twitter:card" content="summary_large_image"/>
    <meta name="twitter:creator" content="@SandraGodSilva"/>
    <meta name="twitter:site" content="@SandraGodSilva"/>
    <meta property="og:title" content="Intro 1"/>
    <meta property="og:description" content="Intro"/>
    <meta property="og:url" content="https://sandragodinhosilva/microbiomes2021/slides/day-01-basics.html#1"/>
    <meta property="og:image" content="https://raw.githubusercontent.com/sandragodinhosilva/microbiomes2021/main/slides/day-01-cover-image.png"/>
    <meta property="og:image:alt" content="Title slide"/>
    <meta property="og:type" content="website"/>
    <meta property="og:locale" content="en_US"/>
    <meta property="article:author" content="Sandra Godinho Silva"/>
    <link href="libs/tile-view/tile-view.css" rel="stylesheet" />
    <script src="libs/tile-view/tile-view.js"></script>
    <link href="libs/panelset/panelset.css" rel="stylesheet" />
    <script src="libs/panelset/panelset.js"></script>
    <script src="libs/clipboard/clipboard.min.js"></script>
    <link href="libs/xaringanExtra-clipboard/xaringanExtra-clipboard.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-clipboard/xaringanExtra-clipboard.js"></script>
    <script>window.xaringanExtraClipboard(null, {"button":"Copy Code","success":"Copied!","error":"Press Ctrl+C to Copy"})</script>
    <script src="libs/xaringanExtra-webcam/webcam.js"></script>
    <script id="xaringanExtra-webcam-options" type="application/json">{"width":"200","height":"200","margin":"1em"}</script>
    <script src="libs/js-cookie/js.cookie.js"></script>
    <script src="libs/peerjs/peerjs.min.js"></script>
    <script src="libs/tiny.toast/toast.min.js"></script>
    <link href="libs/xaringanExtra-broadcast/broadcast.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-broadcast/broadcast.js"></script>
    <link href="libs/shareon/shareon.min.css" rel="stylesheet" />
    <script src="libs/shareon/shareon.min.js"></script>
    <link href="libs/xaringanExtra-shareagain/shareagain.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-shareagain/shareagain.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="slides-style.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Bioinformatic tools for Genome annotation
## Course: Microbiomes (2021)
### Sandra Godinho Silva
### Instituto Superior Técnico
### -04-2021

---






name: about-me
class: center, middle

## About me: Sandra Godinho Silva

&lt;img style="border-radius: 50%;" src="https://sandragodinhosilva.netlify.app/authors/admin/avatar_hu27f7dc34c9db8fd2b063f3fba6be9864_589773_270x270_fill_q75_lanczos_center.jpg"/&gt;

## 

### PhD Student @ Institute for Bioengineering and Biosciences

.fade[Instituto Superior Técnico&lt;br&gt;Lisbon, Portugal]

---
class:  center, middle

# Bioinformatics

Bioinformatics is the science that conjugates  
**biology** + **statistics** + **computer sciences**  
to study biological issues through the analysis of data.

---
class: middle

**Bioinformatics is a constantly changing and updating field.**

Fast developments in:
- software
- computing hardware 
- high throughput technologies 

--

are creating a significant bottleneck:  
**How to thoroughly analyze this massive amount of data?**

---
**How to thoroughly analyze this massive amount of data?**

&lt;img src="img/bigdata_lifesciences.png" width="70%" style="display: block; margin: auto;" /&gt;
.center[
[Stephens et a. PLoS Biology, 2015.](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002195)
]

---
## Integrative OMICS
&lt;img src="img/integrative_omics.png" width="80%" style="display: block; margin: auto;" /&gt;
.center[
[ Yugi et al., Trends Biotechnol. 2016 Apr; 34(4):276–290](https://doi.org/10.1016/j.tibtech.2015.12.013)
]

---
## Choose the most appropriate route for you

Nowadays, bioinformaticians and researchers working in life scientists can choose from an overwhelming collection of exciting technologies and programming languages

Several options:
- Web-based platforms
- Scripting/programming

---
# Web-based platforms

Allow **accessible**, **reproducible**, and **transparent computational research.**

.panelset[

.panel[.panel-name[KBase]

.pull-left[
&lt;img src="img/KBase-logo.png" width="60%" /&gt;
]
.pull-right[The U.S. Department of Energy Systems Biology Knowledgebase [KBase](https://www.kbase.us/)  is an open-source software and data platform designed to meet the grand challenge of systems biology — predicting and designing biological function from the biomolecular (small scale) to the ecological (large scale)
]]


.panel[.panel-name[Galaxy]
.pull-left[
&lt;img src="img/galaxy.png" width="60%" /&gt;
]
.pull-left[Main goal: to develop and maintain a system that enables researchers without informatics expertise to perform computational analyses through the web. A user interacts with Galaxy through the web by uploading and analyzing the data. Galaxy interacts with underlying computational infrastructure (servers that run the analyses and disks that store the data) without exposing it to the user.
(https://galaxyproject.org/)
]]
]


---
# Scripting &amp; programming

.panelset[

.panel[.panel-name[Bash]
.pull-left[
&lt;img src="img/bash.png" width="60%" /&gt;
]
.pull-right[
* Default login shell for most Linux distributions;
* Simple bash scripts are really useful for data manipulation.
]
]

.panel[.panel-name[Python]
.pull-left[
&lt;img src="img/python.png" width="50%" /&gt;
]
.pull-right[
* Clear and powerful object-oriented programming language;
* Portable – runs just about anywhere;
* Clear syntax – relatively easy to learn.
Source: [https://www.python.org/](https://www.python.org)]
]

.panel[.panel-name[Biopython]

.pull-left[
&lt;img src="img/Biopython_logo.svg" width="70%" /&gt;
]
.pull-right[
* Collection of .blue[Python] modules that provide functions to deal with DNA, RNA &amp; protein sequence operations;
* It has sibling projects like BioPerl, BioJava and BioRuby.
Source: [https://biopython.org/](https://biopython.org)
]
]

.panel[.panel-name[R]
.pull-left[
&lt;img src="img/Rlogo.png" width="70%" /&gt;
]
.pull-right[
* .purple[R] is a powerful, popular open-source scripting language
* More than 20 years old - fairly mature - and growing in popularity. 
]
]

]

---

# And much more!

&lt;img src="img/programming_languages.jpeg" width="70%" style="display: block; margin: auto;" /&gt;


---
# Why is R so popular?



---

# R Studio

* Integrated Development Environment (IDE). 
* Provides a graphical interface to R, making it more user-friendly, and providing dozens of useful features.

.pull-left[
&lt;img src="img/rstudio.png" width="100%" /&gt;
]
.pull-right[
.midi[Source: [https://biopython.org/](https://biopython.org)
]
]


---
# Genome Annotation

The process of decorating the genome with information about the localization and function of genes  
The genome annotation procedure will also embrace other functional elements such as promoter sequences,
non-coding RNAs, among others.

---

&lt;img src="img/genome-annotation.jpg" width="70%" style="display: block; margin: auto;" /&gt;
.midi[Source: https://www.ssbs.edu.in/genome-annotation.html]

???
Genome Annotation is the process to identify, analyze and interpret the relevant information related to the raw DNA sequences. Genome Annotation helps to extract structural location and biological functions of genes. Haemophilus influenza was the first free-living bacterium to be decoded in 1995 by Dr. Owen White. The assembling of the quality reads with a reference genome or de novo assembly in order to obtain the complete genome is a primary requirement for Genome Annotation.

---
## Introduction into File Formats
.panelset[

.panel[.panel-name[FASTA]

DNA and protein sequences can be written in FASTA format.
First line:"&gt;" followed by the description. 
In the second line the sequence starts.

&lt;img src="img/fasta.png" width="50%" style="display: block; margin: auto;" /&gt;

]

.panel[.panel-name[GFF3]

General feature format (gene-finding format, generic feature format, GFF) is a file format used for describing genes and other features of DNA, RNA and protein sequences.

&lt;img src="img/gff3_format.png" width="70%" style="display: block; margin: auto;" /&gt;
]


.panel[.panel-name[GENBANK]

The genbank sequence format is a rich format for storing sequences and associated annotations.

&lt;img src="img/gb_full.png" width="50%" style="display: block; margin: auto;" /&gt;

]
]


???
Source: https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html

--
---


# Functional Annotation

Functional gene annotation means the description of the biochemical and biological function of proteins. 

Possible analyses to annotate genes can be:

* similarity searches
* gene cluster prediction for secondary metabolites
* identification of transmembrane domains in protein sequences
* finding gene ontology terms
* pathway information


E.g.: For similarity searches we use NCBI BLAST+ blastp to find similar proteins in a protein database.

---

# Databases

.panelset[

.panel[.panel-name[PFAM]
.left-column[

.center[
&lt;img src="img/pfam.gif" width="100%" style="display: block; margin: auto;" /&gt;
]
]

.right-column[
Pfam is a large collection of protein families, represented by multiple sequence alignments and hidden Markov models (HMMs).
]
]

.panel[.panel-name[COG]

.left-column[
# COG
]
.right-column[
**Clusters of Orthologous Groups (COGs)**

Database of proteins generated by comparing the protein sequences of complete genomes. Each cluster contains groups of paralogs from at least three lineages.

]
]

.panel[.panel-name[Kegg]
.left-column[
&lt;img src="img/KEGG_database_logo.gif" width="100%" /&gt;
]
.right-column[
**Kyoto Encyclopedia of Genes and Genomes**

Database resource for understanding high-level functions and utilities of biological systems, such as the cell, the organism and the ecosystem, from molecular-level information.
]
]

.panel[.panel-name[CAZymes]
.left-column[
&lt;img src="img/cazymes.png" width="150%" /&gt;
]
.right-column[
**Carbohydrate-Active enZymes**

Enzymes involved in the synthesis, metabolism, and transport of carbohydrates. 

CAZymes are organized in families in the continuously updated database [CAZy](www.cazy.org).

Includes: glycoside hydrolases (GHs), glycosyltransferases (GTs), polysaccharide lyases (PLs), carbohydrate esterases (CEs) and carbohydrate binding modules (CBMs). 

]
]

]

---
# Tools for genome annotation


Prokka

[Tutorial](https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html)
---

&lt;img src="img/melange.jpeg" width="100%" /&gt;


Documentation: https://sandragodinhosilva.github.io/MeLanGE/
---
# Your turn!

Genome annotation with Protein Families Database (PFAM)

Tools: WebMGA + R 

http://weizhong-lab.ucsd.edu/webMGA/server/pfam/

---
# WebMGA
&lt;img src="img/webmga_pfam.png" width="55%" style="display: block; margin: auto;" /&gt;
[Link](http://weizhong-lab.ucsd.edu/webMGA/server/)

---

## R script to join annotations from different MAGs



---

class: center, middle

# Thanks!

Here's where you can find me:


Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "solarized-light",
"highlightLines": true,
"countIncrementalSlides": false,
"ratio": "16:9",
"slideNumberFormat": "<div class=\"progress-bar-container\">\n  <div class=\"progress-bar\" style=\"width: calc(%current% / %total% * 100%);\">\n  </div>\n</div>"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
