<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Bioinformatic tools for Genome annotation</title>
    <meta charset="utf-8" />
    <meta name="author" content="Sandra Godinho Silva" />
    <meta name="date" content="2021-04-06" />
    <script src="try_files/header-attrs/header-attrs.js"></script>
    <link href="try_files/tile-view/tile-view.css" rel="stylesheet" />
    <script src="try_files/tile-view/tile-view.js"></script>
    <link href="try_files/panelset/panelset.css" rel="stylesheet" />
    <script src="try_files/panelset/panelset.js"></script>
    <script src="try_files/clipboard/clipboard.min.js"></script>
    <link href="try_files/xaringanExtra-clipboard/xaringanExtra-clipboard.css" rel="stylesheet" />
    <script src="try_files/xaringanExtra-clipboard/xaringanExtra-clipboard.js"></script>
    <script>window.xaringanExtraClipboard(null, {"button":"Copy Code","success":"Copied!","error":"Press Ctrl+C to Copy"})</script>
    <script src="try_files/xaringanExtra-webcam/webcam.js"></script>
    <script id="xaringanExtra-webcam-options" type="application/json">{"width":"200","height":"200","margin":"1em"}</script>
    <script src="try_files/js-cookie/js.cookie.js"></script>
    <script src="try_files/peerjs/peerjs.min.js"></script>
    <script src="try_files/tiny.toast/toast.min.js"></script>
    <link href="try_files/xaringanExtra-broadcast/broadcast.css" rel="stylesheet" />
    <script src="try_files/xaringanExtra-broadcast/broadcast.js"></script>
    <link href="try_files/shareon/shareon.min.css" rel="stylesheet" />
    <script src="try_files/shareon/shareon.min.js"></script>
    <link href="try_files/xaringanExtra-shareagain/shareagain.css" rel="stylesheet" />
    <script src="try_files/xaringanExtra-shareagain/shareagain.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Bioinformatic tools for Genome annotation
## Course: Microbiomes (2021)
### Sandra Godinho Silva
### Instituto Superior Técnico
### 2021-04-06

---






name: about-me
class: center, middle

## About me: Sandra Godinho Silva

&lt;img style="border-radius: 50%;" src="https://sandragodinhosilva.netlify.app/authors/admin/avatar_hu27f7dc34c9db8fd2b063f3fba6be9864_589773_270x270_fill_q75_lanczos_center.jpg"/&gt;

## 

### PhD Student @ Institute for Bioengineering and Biosciences

.fade[Instituto Superior Técnico&lt;br&gt;Lisbon, Portugal]

---
class:  center, middle

# Bioinformatics

Bioinformatics is the science that conjugates  
**biology** + **statistics** + **computer sciences**  
to study biological issues through the analysis of data.

---
class: middle

**Bioinformatics is a constantly changing and updating field.**

Fast developments in:
- software
- computing hardware 
- high throughput technologies 

--

are creating a significant bottleneck:  
**How to thoroughly analyze this massive amount of data?**

---
.center[**How to thoroughly analyze this massive amount of data?**]

&lt;img src="img/bigdata_lifesciences.png" width="70%" style="display: block; margin: auto;" /&gt;
.center[
[Stephens et a. PLoS Biology, 2015.](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002195)
]

---
## Integrative OMICS
&lt;img src="img/integrative_omics.png" width="80%" style="display: block; margin: auto;" /&gt;
.center[
[ Yugi et al., Trends Biotechnol. 2016 Apr; 34(4):276–290](https://doi.org/10.1016/j.tibtech.2015.12.013)
]

---
## Choose the most appropriate route for you

Nowadays, bioinformaticians and researchers working in life scientists can choose from an overwhelming collection of exciting technologies and programming languages.

Several options:
- Web-based platforms
- Scripting/programming

---
# Web-based platforms

Allow **accessible**, **reproducible**, and **transparent computational research.**

.panelset[

.panel[.panel-name[KBase]

.left-column[
&lt;img src="img/KBase-logo.png" width="100%" /&gt;
]
.right-column[The U.S. Department of Energy Systems Biology Knowledgebase [KBase](https://www.kbase.us/)  is an open-source software and data platform designed to meet the grand challenge of systems biology — predicting and designing biological function from the biomolecular (small scale) to the ecological (large scale).
]]


.panel[.panel-name[Galaxy]
.left-column[
&lt;img src="img/galaxy.png" width="100%" /&gt;
]
.right-column[Galaxy provides a system that enables researchers without informatics expertise to perform computational analyses through the web. A user interacts with Galaxy through the web by uploading and analyzing the data. Galaxy interacts with underlying computational infrastructure (servers that run the analyses and disks that store the data) without exposing it to the user.
(https://galaxyproject.org/)
]]
]


---
# Scripting &amp; programming

.panelset[

.panel[.panel-name[Bash]
.left-column[
&lt;img src="img/bash.png" width="100%" /&gt;
]
.right-column[
* Default login shell for most Linux distributions;
* Simple bash scripts are really useful for data manipulation.
]
]

.panel[.panel-name[Python (1/2)]
.left-column[
&lt;img src="img/python.png" width="80%" /&gt;
]
.right-column[
* Clear and powerful object-oriented programming language;
* Portable – runs just about anywhere;
* Clear syntax – relatively easy to learn.
Source: [https://www.python.org/](https://www.python.org)]
]

.panel[.panel-name[Python (2/2)]

.left-column[
&lt;img src="img/Biopython_logo.svg" width="100%" /&gt;
]
.right-column[
## Biopython
* Collection of .blue[Python] modules that provide functions to deal with DNA, RNA &amp; protein sequence operations;
* It has sibling projects like BioPerl, BioJava and BioRuby.
Source: [https://biopython.org/](https://biopython.org)
]
]

.panel[.panel-name[R (1/3)]
.left-column[
&lt;img src="img/Rlogo.png" width="100%" /&gt;
]
.right-column[
* .purple[R] is a powerful, popular open-source scripting language
* More than 20 years old - fairly mature - and growing in popularity. 
]
]


.panel[.panel-name[R (2/3)]
## Why is R so popular?
]

.panel[.panel-name[R (3/3)]
.left-column[
&lt;img src="img/rstudio.png" width="100%" /&gt;
]
.right-column[
## R Studio
* Integrated Development Environment (IDE). 
* Provides a graphical interface to R, making it more user-friendly, and providing dozens of useful features.
.midi[Source: [https://www.rstudio.com/](https://www.rstudio.com/)
]
]


]


]

---

# And much more!

&lt;img src="img/programming_languages.jpeg" width="70%" style="display: block; margin: auto;" /&gt;


---
class: inverse, center, middle

## Break
### **Any questions?**
---
# Genome Annotation

Genome annotation is the process of attaching biological information to sequences.  

Consists of two main steps: 
- gene prediction - identifying elements on the genome;
- attaching biological information to these elements. 

???
Gene annotation is the plotting of genes onto genome assemblies, and indexing their genomic coordinates.
DNA annotation or genome annotation is the process of identifying the locations of genes and all of the coding regions in a genome and determining what those genes do.
---
## Key points:
* Once a genome is sequenced, all of the sequencings must be analyzed to understand what their meaning.
* Critical to annotation is the identification of the genes in a genome, the structure of the genes, and the proteins they encode.
* Once a genome is annotated, further work is done to understand how all the annotated regions interact with each other.

???


 
---
&lt;img src="img/genome-annotation.jpg" width="70%" style="display: block; margin: auto;" /&gt;
.center[Source: https://www.ssbs.edu.in/genome-annotation.html]

???
Genome Annotation is the process to identify, analyze and interpret the relevant information related to the raw DNA sequences. Genome Annotation helps to extract structural location and biological functions of genes. Haemophilus influenza was the first free-living bacterium to be decoded in 1995 by Dr. Owen White. The assembling of the quality reads with a reference genome or de novo assembly in order to obtain the complete genome is a primary requirement for Genome Annotation.

---
# Structural Annotation
The process of identifying genomic elements such as: 
* open reading frames (ORFs) and their localization;
* gene structure;
* coding regions;
* regulatory motifs.

???
Genome annotation is thus the process of attaching
biological information to the genome sequences and starts by identifying the positions of structural
genomic elements, like genes, exons, introns, repeated regions, promoters, etc
---

# Functional Annotation

The process of attaching biological information to genomic elements by describing the the biochemical and biological function of proteins. 

Possible analysis:

* similarity searches
* gene cluster prediction for secondary metabolites
* identification of transmembrane domains in protein sequences
* finding gene ontology terms
* pathway information


E.g.: For similarity searches we use NCBI BLAST+ blastp to find similar proteins in a protein database.

---
# Process Annotation


---
# Homologs, Orthologs, and Paralogs

* **Homologous gene** (homolog): **a gene inherited in two species by a common ancestor**.  



&lt;img src="img/zinc-finger-seq-alignment2.png" width="70%" style="display: block; margin: auto;" /&gt;
Homologous DNA: sequence alignment of a homologous protein from two different species The “*” represent a conserved amino acid in the two proteins. [Source](https://s3-us-west-2.amazonaws.com/courses-images/wp-content/uploads/sites/1950/2017/05/31184035/zinc-finger-seq-alignment2.png)

---
While homologous genes can be similar in sequence, similar sequences are not necessarily homologous.
* Orthologous are homologous genes where a gene diverges after a speciation event, but the gene and its main function are conserved.
* If a gene is duplicated in a species, the resulting duplicated genes are paralogs of each other, even though over time they might become different in sequence composition and function.


???
Homology forms the basis of organization for comparative biology.
A homologous trait is often called a homolog (also spelled homologue). In genetics, the term “homolog” is used both to refer to a homologous protein and to the gene ( DNA sequence) encoding it. 

As with anatomical structures, homology between protein or DNA sequences is defined in terms of shared ancestry. 

Two segments of DNA can have shared ancestry because of either a speciation event (orthologs) or a duplication event (paralogs). 

Homology among proteins or DNA is often incorrectly concluded on the basis of sequence similarity. The terms “percent homology” and “sequence similarity” are often used interchangeably. As with anatomical structures, high sequence similarity might occur because of convergent evolution, or, as with shorter sequences, because of chance. Such sequences are similar, but not homologous. Sequence regions that are homologous are also called conserved. This is not to be confused with conservation in amino acid sequences in which the amino acid at a specific position has been substituted with a different one with functionally equivalent physicochemical properties. One can, however, refer to partial homology where a fraction of the sequences compared (are presumed to) share descent, while the rest does not. For example, partial homology may result from a gene fusion event.
---

---

# Tools for genome annotation




???
Automatic annotation tools try to perform all of this by computer analysis, as opposed to manual annotation (a.k.a. curation) which involves human expertise. Ideally, these approaches co-exist and complement each other in the same annotation pipeline (process). 
The basic level of annotation is using BLAST for finding similarities, and then annotating genomes based on that. However, nowadays more and more additional information is added to the annotation platform. The additional information allows manual annotators to deconvolute discrepancies between genes that are given the same annotation. 

Some databases use genome context information, similarity scores, experimental data, and integrations of other resources to provide genome annotations through their Subsystems approach. Other databases rely on both curated data sources as well as a range of different software tools in their automated genome annotation pipeline.

https://bio.libretexts.org/Bookshelves/Microbiology/Book%3A_Microbiology_(Boundless)/7%3A_Microbial_Genetics/7.13%3A_Bioinformatics/7.13B%3A_Annotating_Genomes
---
Prokka

[Tutorial](https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html)
---
# Databases

.panelset[

.panel[.panel-name[PFAM]
.left-column[
.center[
&lt;img src="img/pfam.gif" width="100%" style="display: block; margin: auto;" /&gt;
]
]
.right-column[
Pfam is a large collection of protein families, represented by multiple sequence alignments and hidden Markov models (HMMs).  
Source: [http://pfam.xfam.org/](http://pfam.xfam.org/)
]
]

.panel[.panel-name[COG]
.left-column[
# COG
]
.right-column[
**Clusters of Orthologous Groups (COGs)**
Database of proteins generated by comparing the protein sequences of complete genomes. Each cluster contains groups of paralogs from at least three lineages.  
Source: [https://www.ncbi.nlm.nih.gov/research/cog-project/](https://www.ncbi.nlm.nih.gov/research/cog-project/)
]
]

.panel[.panel-name[Kegg]
.left-column[
&lt;img src="img/KEGG_database_logo.gif" width="100%" /&gt;
]
.right-column[
**Kyoto Encyclopedia of Genes and Genomes**
Database resource for understanding high-level functions and utilities of biological systems, such as the cell, the organism and the ecosystem, from molecular-level information.  
Source: [https://www.genome.jp/kegg/](https://www.genome.jp/kegg/)
]
]

.panel[.panel-name[CAZymes]
.left-column[
&lt;img src="img/cazymes.png" width="150%" /&gt;
]
.right-column[
**Carbohydrate-Active enZymes**
Enzymes involved in the synthesis, metabolism, and transport of carbohydrates. 
CAZymes are organized in families in the continuously updated database [CAZy](www.cazy.org).
Source: [http://www.cazy.org/](http://www.cazy.org/)
???
Includes: glycoside hydrolases (GHs), glycosyltransferases (GTs), polysaccharide lyases (PLs), carbohydrate esterases (CEs) and carbohydrate binding modules (CBMs). 
]
]

]
---
## Introduction to File Formats
.panelset[

.panel[.panel-name[FASTA]

DNA and protein sequences can be written in FASTA format.
First line:"&gt;" followed by the description. 
In the second line the sequence starts.

&lt;img src="img/fasta.png" width="50%" style="display: block; margin: auto;" /&gt;

]

.panel[.panel-name[GFF3]

General feature format (gene-finding format, generic feature format, GFF) is a file format used for describing genes and other features of DNA, RNA and protein sequences.

&lt;img src="img/gff3_format.png" width="70%" style="display: block; margin: auto;" /&gt;
]


.panel[.panel-name[GENBANK]

The genbank sequence format is a rich format for storing sequences and associated annotations.

&lt;img src="img/gb_full.png" width="50%" style="display: block; margin: auto;" /&gt;

]
]


???
Source: https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html


---

&lt;img src="img/melange.jpeg" width="100%" /&gt;


Documentation: https://sandragodinhosilva.github.io/MeLanGE/
---
# Your turn!

Genome annotation with Protein Families Database (PFAM)

Tools: WebMGA + R 

http://weizhong-lab.ucsd.edu/webMGA/server/pfam/

---
# WebMGA
&lt;img src="img/webmga_pfam.png" width="55%" style="display: block; margin: auto;" /&gt;
[Link](http://weizhong-lab.ucsd.edu/webMGA/server/)

---

## R script to join annotations from different MAGs



---

class: center, middle

# Thanks!

Here's where you can find me:


Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).
---
class: center, middle

# Thanks!

Slides created via the R packages:

[**xaringan**](https://github.com/yihui/xaringan)&lt;br&gt;
[gadenbuie/xaringanthemer](https://github.com/gadenbuie/xaringanthemer)

The chakra comes from [remark.js](https://remarkjs.com), [**knitr**](http://yihui.name/knitr), and [R Markdown](https://rmarkdown.rstudio.com).
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%",
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
